astro/src/content/quickstarts/quickstart-javascript-nextjs-web.mdx

```diff
@@ -0,0 +1,283 @@
+
+`src/components/login-button.tsx`
+<RemoteCode url={frontmatter.coderoot + "/complete-application/src/components/login-button.tsx"}
+  lang="ts" />
+
+#### Layout
+
+<Aside type="tip">
+If this is your first time using Next.js's App Router, I would highly suggest reading through at least the <a href="https://nextjs.org/docs/app/building-your-application/routing" target="_blank">Routing Fundamentals</a>
+</Aside>
+
+Below you will find the full code for the [Root Layout](https://nextjs.org/docs/app/building-your-application/routing/pages-and-layouts#root-layout-required)
+
+This has the overall structure of our application, the other pages will be added where `{children}` is located.
+
+There are two items of note:
+1. Getting the session information - `const session = await getServerSession(authOptions);`
+1. Passing the session to the Login Button - `<LoginButton session={session} />`
+
+`src/app/layout.tsx`
+<RemoteCode url={frontmatter.coderoot + "/complete-application/src/app/layout.tsx"}
+  lang="ts" />
+
+#### Home Page
+
+Create a new file `src/app/page.tsx` which will have the Homepage details. Not much here just an image and another Login button.
+
+`src/app/page.tsx`
+<RemoteCode url={frontmatter.coderoot + "/complete-application/src/app/page.tsx"}
+  lang="ts" />
+
+
+#### Account Page (protected)
+
+Create a new file `src/app/account/page.tsx` which will have the Account details.
+
+One special note here is that we check to see if the session is missing we redirect back to the homepage, which protects this page on the server for unauthorized access. (You can find the same when a user is logged in on the homepage, it will redirect to `/account`)
+
+```ts
+if (!session) {
+  redirect('/');
+}
+```
+
+`src/app/account/page.tsx`
+<RemoteCode url={frontmatter.coderoot + "/complete-application/src/app/account/page.tsx"}
+  lang="ts" />
+
+### Run the App!
+Run your Next.js application. 
+
+<Aside type="caution">
+If you did the `Full Example` make sure you are in the directory `complete-application` if you did the `Walkthrough Example` make sure you are in the `changebank` directory.
+</Aside>
+
+
+```shell
+npm run dev
+```
+
+Try clicking the `Login` button at the top or center of the screen.
+
+![Changebank Home Page](/img/articles/quickstart-javascript-nextjs-web/homepage.png)
+
+
+This will take you through the `NextAuth.js` authentication flow. First prompting you to select `Sign in with FusionAuth`. 
+
+![Changebank NextAuth.js prompt](/img/articles/quickstart-javascript-nextjs-web/nextauth.png)
+
+You can then login to FusionAuth with Email: `richard@example.com` Password: `password` (as you might expect not ideal for production.)
+
+![FusionAuth Login Form](/img/articles/quickstart-javascript-nextjs-web/fusionauth-login.png)
+
+This will then take you back to the application in the API route it will check for a session and appropriately redirect you to the `/account` route when your session has been establised.
```

Comment:
```suggestion
This will then take you back to the application in the API route it will check for a session and appropriately redirect you to the `/account` route when your session has been established.
```